morango/utils/morango_mptt.py

```diff
@@ -1,40 +1,29 @@
-import uuid
+from mptt import managers, models, querysets
+from morango.models import SyncableModelQuerySet
 
-from mptt import managers, models
-from morango.models import SyncableModel
 
-from .uuids import UUIDField
+class MorangoTreeQuerySet(querysets.TreeQuerySet, SyncableModelQuerySet):
+    pass
 
 
-class MorangoTreeManager(managers.TreeManager):
-    # Override the logic in Django MPTT that assigns tree ids, as we need to ensure tree ids do not
-    # conflict for facilities created on differentdevices and then later synced to the same device
-    # By default, tree ids are auto-increasing integers, but we use UUIDs to avoid collisions.
-    def _get_next_tree_id(self):
-        return uuid.uuid4().hex
+class MorangoMPTTTreeManager(managers.TreeManager):
+
+    def get_queryset(self):
+        return MorangoTreeQuerySet(self.model, using=self._db)
+
+    @managers.delegate_manager
+    def _mptt_update(self, qs=None, **items):
+        if qs is None:
+            qs = self
+        return qs.update(dirty_bit_signal=None, **self._translate_lookups(**items))
```

Comment:
Should work to just add the dirty bit argument to the kwargs and call `super`, rather than duplicating the logic here. Won't need to add the decorator here then either.